test/jit/test_profiler.py

```diff
@@ -31,6 +31,29 @@ def tearDown(self):
         torch._C._jit_set_texpr_fuser_enabled(self.texpr_fuser_state)
         torch.set_default_dtype(self.default_dtype)
 
+    def test_tensor_type_not_determined_by_inputs(self):
+        @torch.jit.script
+        def scalar_type_input(x, y, z):
+            return x + y + 4 + z.item()
+
+        x = torch.tensor([2, 2])
+        scalar_type_input(x, x, torch.tensor(1))
+        scalar_type_input(x, x, torch.tensor(1))
+        scalar_type_input(x, x, torch.tensor(1.0))
+        g = torch.jit.last_executed_optimized_graph()
+        FileCheck().check("TensorExpr").check("Scalar = aten::item").check_next("Tensor = aten::add").check("TensorExpr").run(g)
```

Comment:
Nit: might be useful to have a comment describing what IR we're expecting to get and why. Also, the test could be made more targeted if written as in `test_te_fuser_pass.cpp` where the fuser pass is invoked explicitly.